### ISSUE TITLE
Ability to change severity

### ISSUE THREAD

**Issue by SteveAbb** [DependencyTrack/dependency-track#581]
### Current Behavior:
Currently, it is only possible to mark a vulnerability as a false positive or exploitable. 

### Proposed Behavior:
Give the user the ability to change the severity of a given vulnerability. Perhaps this could be a separate analysis state of "SEVERITY_CHANGE" or something similar. I'm envisioning this being used in situations where environmental or temporal factors change the severity.

**Comment by stevespringett** [comment]
Changing severity will not be supported. Those are derived from the vulnerability itself. However, adding the impact or risk of a vulnerable component on a given application, would be a welcome enhancement.

This is related to conversations happening in #357. Along with the 'remediation guidance' field, it may be useful to have an 'impact' or 'risk' field with critical, high, medium, and low as potential options.

**Comment by SteveAbb** [comment]
Yes, I was thinking of changing the risk/impact of a component in the context of a given application, not the original severity of the vulnerability.

I was thinking about how the CVSS 3.1 user guide talks about this in section 3.7:

https://www.first.org/cvss/user-guide

> When scoring a vulnerability in a given implementation using the impacted library, the score must be re-calculated for that specific implementation. For example, if an implementation embeds the vulnerable library mentioned in the previous example, but only operates on local files, the Attack Vector (AV) would be Local (L).

**Comment by stevespringett** [comment]
Thanks for the link.

DT already has a graphical CVSS calculator. It's only used for the private vuln repo, but it's there.
https://docs.dependencytrack.org/datasources/private-vuln-repo/

Not all vulnerabilities have CVSS scores - NPM Advisories do not. Depending on the audience, CVSS may be viewed in a negative way. I personally prefer the OWASP Risk Rating methodology.
https://wiki.owasp.org/index.php/OWASP_Risk_Rating_Methodology
https://security-net.biz/files/owaspriskcalc.html

I've already built Java libraries for CVSS calculation (DT uses this already) and for OWASP Risk Rating (DT does not currently use this), so perhaps the we could give them the choice to use either one.

**Comment by extreme4all** [comment]
seems related to #Status: Open.
#216
